collated/main/A0161247J.md

```diff
@@ -0,0 +1,637 @@
+        }
+        return new ClearCommand(arguments);
+    }
+
+    /**
+     * Returns true if arguments for clear command are valid
+     */
+    private boolean isValidArgs(String arguments) {
+        return arguments.isEmpty() || arguments.equals(ClearCommand.OPTION_CLEAR_ALL);
+    }
+
+}
+```
+###### \java\seedu\task\logic\parser\EditParser.java
+``` java
+package seedu.task.logic.parser;
+
+import static seedu.task.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import java.text.ParseException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import seedu.task.commons.exceptions.IllegalValueException;
+import seedu.task.commons.util.DateUtil;
+import seedu.task.logic.commands.Command;
+import seedu.task.logic.commands.EditCommand;
+import seedu.task.logic.commands.IncorrectCommand;
+import seedu.task.model.task.Name;
+import seedu.task.model.task.TaskDate;
+
+/**
+ * Parser class used to parse an edit command
+ */
+public class EditParser extends Parser{
+	private final Pattern FLOATING_ARGS_FORMAT = Pattern.compile("\\s*(?<index>\\d+)\\s*(?<name>.+)");
+	private final Pattern DEADLINE_ARGS_FORMAT = Pattern.compile("\\s*(?<index>\\d+)\\s*(?<endDate>\\d{2}-\\d{2}-\\d{4})\\s*(?<endTime>\\d{2}:\\d{2})?\\s*");
+	private final Pattern EVENT_ARGS_FORMAT = Pattern.compile("\\s*(?<index>\\d+)\\s*(?<startDate>\\d{2}-\\d{2}-\\d{4})\\s*(?<startTime>\\d{2}:\\d{2})?\\s+(?<endDate>\\d{2}-\\d{2}-\\d{4})\\s*(?<endTime>\\d{2}:\\d{2})?\\s*");
+	
+	/**
+     * Parses arguments in the context of the edit task command.
+     * @param args full command args string
+     * @return the prepared command
+     */
+	@Override
+	public Command parseCommand(String args) {
+		Command toReturn = null;
+		boolean hasException = false;
+		
+		try {
+			if (isEventCommand(args)) {
+				toReturn = createEventTask(args);
+			} else if (isDeadlineCommand(args)) {
+				toReturn = createDeadlineTask(args);
+			} else if (isFloatingCommand(args)) {
+				toReturn = createFloatingTask(args);
+			} else {
+				throw new IllegalArgumentException();
+			}
+		} catch (NullPointerException e) {
+        	hasException = true;
+        } catch (ParseException e) {
+        	hasException = true;
+        } catch (IndexOutOfBoundsException e) {
+        	hasException = true;
+        } catch (IllegalArgumentException e) {
+        	hasException = true;
+        } catch (IllegalValueException e) {
+            hasException = true;
+        } catch (DateTimeParseException e) {
+            hasException = true;
+        }
+		
+		if (hasException) {
+        	toReturn = new IncorrectCommand(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, 
+                    		EditCommand.MESSAGE_USAGE));
+        }
+		
+		return toReturn;
+    }
+	
+	/**
+	 * Method used to retrieve the index from a string argument 
+	 * @param argIndex
```

Comment:
please give a brief description of what the parameter is for; in the current form, there is no additional information this comment line is providing